exe/ReferenceDoc.hs

```diff
@@ -0,0 +1,121 @@
+{-# LANGUAGE QuasiQuotes #-}
+-- | This executable generates the radicle reference docs. This is a markdown
+-- document which contains documentation for all the primitive functions and all
+-- the functions defined in the prelude.
+module ReferenceDoc where
+
+import           Protolude
+
+import           Data.List
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+import qualified GHC.Exts as GhcExts
+import           Radicle
+import           Radicle.Internal.Doc (md)
+import qualified Radicle.Internal.Doc as Doc
+import           System.Console.Haskeline (defaultSettings, runInputT)
+
+main :: IO ()
+main = do
+    res_ <- runInputT defaultSettings $
+             interpret
+               "reference-doc"
+               "(do (load! \"rad/prelude.rad\") (get-current-env))"
+               replBindings
+    res <- res_ `lPanic` "Error running the prelude."
+    s :: Bindings () <- fromRad res `lPanic` "Couldn't convert radicle state."
+    let vars = GhcExts.toList (bindingsEnv s)
+    let undocd = [ fromIdent i | (i, Nothing, _) <- vars ]
+    checkAllDocumented undocd
+    let e = Map.fromList [ (fromIdent i, d) | (i, Just d, _) <- vars ]
+    checkAllInReference e
+    d <- Doc.cleanupPandocMd (doc e) `lPanic` "Invalid markdown."
+    writeFile "docs/source/reference.md" d
+  where
+    doc e =
+      let sec tit fns p = "## " <> tit <> "\n\n" <> p <> "\n\n" <> funs e fns in
+      T.intercalate "\n\n"
+      [ "# Radicle reference"
```

Comment:
You can also use something like [interpolate](http://hackage.haskell.org/package/interpolate-0.2.0/docs/Data-String-Interpolate.html) here to make things a little more readable.